File: pf-core/src/main/java/com/purdynet/util/FileUtil.java
package com.purdynet.util;

import java.io.BufferedReader;
import java.io.FileReader;
import java.util.ArrayList;
import java.util.List;

/**
 * Created with IntelliJ IDEA.
 * User: dnpurdy
 * Date: 9/22/13
 * Time: 3:41 PM
 * To change this template use File | Settings | File Templates.
 */
public class FileUtil
{
    public static List<String> getSymbolsFromFile(String filename)
    {
        List<String> ret = new ArrayList<String>();

        FileReader fis;
        BufferedReader reader = null;
        String line;
        try {
            //Burn header
            fis = new FileReader(filename);
            reader = new BufferedReader(fis);
            while((line = reader.readLine()) != null)
            {
                ret.add(line);
            }
            reader.close();
        }
        catch(Exception e)
        {
            e.printStackTrace();
        }
        return ret;
    }

}

File: pf-core/src/main/scala/com/purdynet/util/FileUtil.scala
package com.purdynet.util

import scala.collection.JavaConverters._
import scala.io.Source
/**
  * Created by dnpurdy on 11/2/16.
  */
object FileUtil {
  def getSymbolsFromFile(filename: String): List[String] = {
    Source.fromFile(filename).getLines.toList
  }

  def getSymbolsFromFileJava(filename: String): java.util.List[String] = {
    getSymbolsFromFile(filename).asJava
  }
}
